### PR TITLE
Frankreed/fix numshot and gesturehandler bug

### PR DESCRIPTION
**Pull request by FrankreedX**
Fix a bunch of small bugs. This branch has been rebased on top of the barchart pr so it can't be merged right now...

### CODE REVIEW

**Comment by solderq35** [comment]
I got the shotnum bug fixed on my branch (https://github.com/Golf-Drill-Challenge-App/Golf-App/tree/solderq35/shotnum-hotfix), although if your branch also works we can just merge this one

File: app/content/assignments/index.js [hunk 1]
```diff
@@ -49,6 +49,22 @@ const AssignmentList = () => {
     );
   }
 
+  // Handle both errors of 'cannot read property "reduce" of undefined' and
```
Comment (solderq35) [inline app/content/assignments/index.js:52]: This code is already existing in this file at line ~36. Is there a reason for this change?

Also I think there is a "login fixes" commit in this PR that is already in layout branch?

**Review by solderq35 — CHANGES_REQUESTED** [review]

**Comment by solderq35** [comment]
Other than the question about the code in assignments/index.js, the code is working as expected on ios and android

**Comment by FrankreedX** [comment]
cleaned up the history

**Review by solderq35 — APPROVED** [review]

**Comment by solderq35** [comment]
Look good to me

**Pull request merged**